File: dcp-solutions/2024-12-13/solution.c
```c
#include <stdlib.h>
#include <limits.h>

int maxPathSum(struct TreeNode* root) {
    int res = INT_MIN;
    dfs(root, 0);
    return res;

}

int dfs(struct TreeNode* node, int cur_sum) {
    if (!node)
        return 0;

    int lsum = max(0, dfs(node->left, cur_sum));
    int rsum = max(0, dfs(node->right, cur_sum));

    cur_sum += node->val;
    res = max(res, cur_sum + lsum + rsum);

    return node->val > 0 ? max(lsum, rsum) : 0;
}
```

File: dcp-solutions/2024-12-13/solution.go
```markdown
// Definition for a binary tree node.
type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func maxPathSum(root *TreeNode) int {
	var res int64 = -1_64
	var maxGain func(node *TreeNode)
	maxGain = func(node *TreeNode) {
		if node == nil {
			return
		}
		maxL := int64(-1e18)
		maxR := int64(-1e18)
		if node.Left != nil {
			l := maxPathSum(node.Left)
			maxL = max(maxL, l+int64(node.Val))
		}
		if node.Right != nil {
			r := maxPathSum(node.Right)
			maxR = max(maxR, r+int64(node.Val))
		}
		res = max(res, int64(max(maxL, maxR)))
		return int64(max(maxL, maxR) + int64(node.Val))
	}
	maxGain(root)
	return int(res)
}

```